inc/Engine/Optimization/DelayJS/Admin/Settings.php

```diff
@@ -91,8 +100,10 @@ public function sanitize_options( $input, $settings ) : array {
 		$input['delay_js_exclusions'] = ! empty( $input['delay_js_exclusions'] ) ? rocket_sanitize_textarea_field( 'delay_js_exclusions', $input['delay_js_exclusions'] ) : [];
 
 		if ( empty( $input['delay_js_exclusions_selected'] ) ) {
-			$input['delay_js_exclusions_selected']            = '';
-			$input['delay_js_exclusions_selected_exclusions'] = '';
+			$input['delay_js_exclusions_selected']            = [];
+			$input['delay_js_exclusions_selected_exclusions'] = [];
+		} else {
+			$input['delay_js_exclusions_selected_exclusions'] = $this->site_list->get_delayjs_items_exclusions( $input['delay_js_exclusions_selected'] );
 		}
```

Comment:
I'd suggest to move this to its own sanitize method in the SiteList class, and we can use the filter to hook the sanitize method